conans/model/requires.py

```diff
@@ -123,9 +126,14 @@ def update(self, down_reqs, output, own_ref, down_ref):
                 # update dependency
                 other_ref = other_req.ref
                 if other_ref and other_ref != req.ref:
-                    output.info("%s requirement %s overridden by %s to %s "
-                                % (own_ref, req.ref, down_ref or "your conanfile",
-                                   other_ref))
+                    msg = "requirement %s overridden by %s to %s " \
+                          % (req.ref, down_ref or "your conanfile", other_ref)
+
+                    if raise_on_override and not other_req.override:
+                        raise ConanException(msg)
+
+                    msg = "%s %s" % (own_ref, msg)
+                    output.info(msg)
```

Comment:
I would like to change this to `output.warn`, this would be more consistent with the idea of promoting _warnings_ to _errors_.

Reply:
agree

Reply:
Ok.